docs/variables.rst

```diff
@@ -0,0 +1,131 @@
+
+    --vars VARS           Variables file
+    --env-vars ENV_VARS   Comma separated list of variables to import from the environment
+
+To import variables from a JSON or YAML formatted file, the `--vars` argument should be used.
+To import environment variables the `--env-vars` argument should be used. It is possible to specify
+multiple environment variables separated by a comma.
+
+.. important::
+    Variables that are supplied using the `--vars` argument will have their data type preserved, but environment
+    variables parsed via `--env-vars` are always treated as strings. This is important to consider when parsing
+    variables into your rulebook.
+
+.. note::
+    In cases where the same variable is supplied via `--env-vars` and `--vars`, the one from
+    `--env-vars` will always take precedence.
+
+
+Accessing variables in your rulebook
+------------------------------------
+
+The are two important considerations to take into account when accessing variables in a rulebook:
+
+    1. In order to use variables in conditions, the variables supplied at runtime will be placed into a namespace
+    called **vars**. You must use the **vars.** prefix followed by the variable name to use variables in conditions.
+
+    2. Jinja substitution should be used when accessing variables in any other part of the rulebook, such as in
+    the source configuration or in actions.
+
+Let's take the following example rulebook:
+
+    .. code-block:: yaml
+
+        ---
+        - name: Testing vars
+          hosts: all
+          sources:
+            - ansible.eda.range:
+                limit: "{{ src_range_limit }}"
+          rules:
+            - name: Say Hello
+              condition: event.i == vars.match_this_int
+              action:
+                echo:
+                  message: "Hi, I'm {{ MY_NAME }}."
+
+In this rulebook we use three different variables. One, **MY_NAME**, we will supply using an environment variable.
+The other two will be supplied in a file called `vars.yml` which looks like this:
+
+    .. code-block:: yaml
+
+        ---
+        src_range_limit: 4
+        match_this_int: 2
+
+To run this rulebook we will need to use both `--vars` and `--env-vars` to parse the variables into the rulebook. Therefore,
+we would execute the rulebook as follows:
+
+    .. code-block:: console
+
+        $ MY_NAME="Bob" ansible-rulebook -i inventory.yml --rulebook hello_there.yml --vars vars.yml --env-vars MY_NAME
+
+Which would return the following output:
+
+    .. code-block:: console
+
+        2023-02-09 13:37:73.311337 : Hi, I'm Bob
+
+.. note::
+    - Quotation marks around Jinja substituted variables are **mandatory** when used as the first element in a value.
+    - Note the variables that use Jinja substitution in the source config and action message, in contrast to
+      the variable in the condition which instead uses the "**vars.**" prefix. Jinja substitution will not work
```

Comment:
There is a note in conditions to remark that Jinja is not supported for conditions. Maybe is good place to link it

https://github.com/ansible/ansible-rulebook/blame/main/docs/conditions.rst#L28